Sources/NIOHPACK/IndexedHeaderTable.swift

```diff
@@ -0,0 +1,150 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+
+/// The unified header table used by HTTP/2, encompassing both static and dynamic tables.
+public struct IndexedHeaderTable {
+    // private but tests
+    let staticTable = HeaderTableStorage(staticHeaderList: StaticHeaderTable)
+    var dynamicTable: DynamicHeaderTable
+    
+    /// Creates a new header table, optionally specifying a maximum size for the dynamic
+    /// portion of the table.
+    ///
+    /// - Parameter maxDynamicTableSize: Maximum size of the dynamic table. Default = 4096.
+    init(maxDynamicTableSize: Int = DynamicHeaderTable.defaultSize) {
+        self.dynamicTable = DynamicHeaderTable(maximumLength: maxDynamicTableSize)
+    }
+    
+    /// Obtains the header key/value pair at the given index within the table.
+    ///
+    /// - note: Per RFC 7541, this uses a *1-based* index.
+    /// - Parameter index: The index to query.
+    /// - Returns: A tuple containing the name and value of the stored header.
+    /// - Throws: `NIOHPACKErrors.InvalidHeaderIndex` if the supplied index was invalid.
+    public func header(at index: Int) throws -> (name: String, value: String) {
```

Comment:
Should this function not just be implemented in terms of the one below?

Reply:
Correct.